### PR TITLE
Fixes #7937: Check and/or enhance ncf usage in initial promises

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/7937

### CODE REVIEW

File: rudder-agent/SOURCES/rudder-agent-postinst [hunk 1]
```diff
@@ -73,6 +73,8 @@ if [ ! -f /var/rudder/cfengine-community/inputs/promises.cf ] || ! /var/rudder/c
 then
   rm -rf /var/rudder/cfengine-community/inputs/* || true
   cp -r /opt/rudder/share/initial-promises/* /var/rudder/cfengine-community/inputs/
+  rm -rf /var/rudder/ncf/common/*
+  cp -r /opt/rudder/share/initial-ncf/* /var/rudder/ncf/common/
 fi
```
Comment (jooooooon) [inline rudder-agent/SOURCES/rudder-agent-postinst]: I think this whole block of code should live in check-rudder-agent (or soon to be "rudder agent check"). The command is already called at the end of this script (the last thing that happens) and it should be doing this reset anyway.

However, that would likely involve changing the order below so that the agent start command comes after the check.

File: rudder-agent/SOURCES/check-rudder-agent [hunk 2]
```diff
@@ -103,9 +103,9 @@ check_and_fix_cfengine_processes() {
     echo " Done"
   fi
 
-  # Get the value of rudder-agent run interval from file /var/rudder/cfengine-community/inputs/run_interval
-  if [ -f "${CFE_DIR}/inputs/run_interval" ]; then
-    RUN_INTERVAL=`cat "${CFE_DIR}/inputs/run_interval"`
+  # Get the value of rudder-agent run interval from file /var/rudder/cfengine-community/inputs/run-interval
+  if [ -f "${CFE_DIR}/inputs/run-interval" ]; then
+    RUN_INTERVAL=`cat "${CFE_DIR}/inputs/run-interval"`
```
Comment (jooooooon) [inline rudder-agent/SOURCES/check-rudder-agent]: Is this supposed to be here? It seems unrelated.

File: rudder-agent/SOURCES/Makefile [hunk 3]
```diff
@@ -276,10 +277,14 @@ rudder.8.gz:
...
 	gzip -f rudder.8
 
-initial-promises: ./rudder-sources
+initial-promises: ./rudder-sources initial-ncf
 	rm -rf ./initial-promises/
 	cp -r ./rudder-sources/rudder-techniques/initial-promises/node-server/ ./initial-promises
 
+initial-ncf: ./rudder-sources
+	rm -rf ./initial-ncf
+	cp -r ./rudder-sources/ncf/tree ./initial-ncf
```
Comment (jooooooon) [inline rudder-agent/SOURCES/Makefile]: For initial-ncf to work, we'll need an initial ncf.conf file too, that should contain the configuration to use log_rudder, and the 5309 CFEngine port (and whatever else we change in there).

File: rudder-agent/SOURCES/check-rudder-agent (hunk 2 above)
Comment (peckpeck) [inline rudder-agent/SOURCES/check-rudder-agent]: The name has been changed in the initial promises merge PRs, since this one is the continuation of those PRs i've chosen to put it here.

File: rudder-agent/SOURCES/rudder-agent-postinst (hunk 1 above)
Comment (peckpeck) [inline rudder-agent/SOURCES/rudder-agent-postinst]: We could put the agent startup code below in check-rudder-agent too.
The only difference is that would not have an install specific message anymore

**Comment by peckpeck** [comment]
Commit modified
Add rudder specific ncf.conf

File: rudder-agent/SOURCES/check-rudder-agent (hunk 2 above)
Comment (jooooooon) [inline rudder-agent/SOURCES/check-rudder-agent]: OK.

File: rudder-agent/SOURCES/rudder-agent-postinst (hunk 1 above)
Comment (jooooooon) [inline rudder-agent/SOURCES/rudder-agent-postinst]: In the case of a "normal" start (agent not disabled + not first install) there is no specific install message, so that's fine by me. We do need to keep the install specific messages when agent is disabled or not yet configured, here, though.

File: rudder-agent/SOURCES/Makefile [hunk 4]
```diff
@@ -276,10 +277,16 @@ rudder.8.gz:
...
+	# retreive the rudder specific ncf.conf
+	cp ./initial-promises/node-server/distributePolicy/ncf/rudder-ncf-conf ./initial-ncf/ncf.conf
```
Comment (peckpeck) [inline rudder-agent/SOURCES/Makefile]: Added here

File: rudder-agent/SOURCES/check-rudder-agent (hunk 2 above)
Comment (peckpeck) [inline rudder-agent/SOURCES/check-rudder-agent]: I will do this in another PR to make it easier to read/merge.
See http://www.rudder-project.org/redmine/issues/8814

Comment (peckpeck) [inline rudder-agent/SOURCES/check-rudder-agent]: Sorry, this comment belongs to the thread below, about moving things to check-rudder-agent

File: rudder-agent/SOURCES/Makefile (hunk 4 above)
Comment (jooooooon) [inline rudder-agent/SOURCES/Makefile]: :+1:

File: rudder-agent/SOURCES/check-rudder-agent (hunk 2 above)
Comment (jooooooon) [inline rudder-agent/SOURCES/check-rudder-agent]: OK
